src/main/java/org/opentripplanner/graph_builder/issue/api/IssueWithSource.java

```diff
@@ -0,0 +1,32 @@
+package org.opentripplanner.graph_builder.issue.api;
+
+public class IssueWithSource implements DataImportIssue {
+
+  private final String type;
+  private final String message;
+  private final String source;
+
+  private IssueWithSource(String type, String message, String source) {
+    this.type = type;
+    this.message = message;
+    this.source = source;
+  }
+
+  public static IssueWithSource issue(String type, String message, String source) {
+    return new IssueWithSource(type, message, source);
+  }
+
+  public static IssueWithSource issue(DataImportIssue issue, String source) {
+    return new IssueWithSource(issue.getType(), issue.getMessage(), source);
+  }
```

Comment:
I only think you need the constructor - you call this from one place?

Reply:
Yes we do